### PR TITLE
add cluster domain

### PR DESCRIPTION
**Pull request by wenchajun**
Add FluentBit Cluster CRD such as clusterinput.fluentbit.fluent.io while keeping the original namespace crd like Input.logging.kubesphere.io

This is another form of adding a domain
Signed-off-by: chengdehao <dehaocheng@yunify.com>

### CODE REVIEW

**Comment by benjaminhuo** [comment]
> Add FluentBit Cluster CRD such as clusterinput.fluentbit.fluent.io while keeping the original namespace crd like Input.logging.kubesphere.io
> 
> This is another form of adding a domain Signed-off-by: chengdehao [dehaocheng@yunify.com](mailto:dehaocheng@yunify.com)

This PR is similar to https://github.com/fluent/fluentbit-operator/pull/167 ? A better approach to do the same thing?

File: pkg/operator/daemonset.go [hunk 1]
```diff
@@ -184,6 +192,11 @@ func MakeDaemonSet(fb v1alpha2.FluentBit, logPath string) appsv1.DaemonSet {
 									ReadOnly:  true,
 									MountPath: "/fluent-bit/config",
 								},
+								{
+									Name:      "clusterconfig",
+									ReadOnly:  true,
+									MountPath: "/fluent-bit/clusterconfig",
+								},
```
Comment (benjaminhuo) [inline pkg/operator/daemonset.go:199]: Is it ok to have two config here? 
one local, one cluster？

**Comment by wenchajun** [comment]
Yes, although this PR implements the same functionality as the original PR, it is implemented in a different way. 
There are three main differences.
- [The secrets of the namespace](https://github.com/fluent/fluentbit-operator/pull/170/files#diff-0b752120e79a0e2d3a24281bbe4463ba6c9b115c9c0db57433160cb32fcf1f51R109-R110) In this PR, cluster-level secret is generated in the namespace to which the operator belongs, not in every namespace. Maybe in the future, you can specify namespaces to generate the corresponding secret?
- [cluster and namespace config](https://github.com/fluent/fluentbit-operator/pull/170/files#diff-402c52300767eca3e94946b162473b45fcf6971a390788a2973b661bbc01c311L1)In the https://github.com/fluent/fluentbit-operator/pull/167,You can bind secret, cluster-fluentbit-config, or Fluentbit-config.like this

In this PR, you need to bind namespace-level Config to cluster-level configs, like this

Frankly speaking, I think both have their advantages and disadvantages. As for the choice, maybe we can discuss it

File: conf/fluent-bit.conf [hunk 2]
```diff
@@ -1 +1,2 @@
 @INCLUDE /fluent-bit/config/fluent-bit.conf
+@INCLUDE /fluent-bit/clusterconfig/cluster-fluent-bit.conf
```
Comment (wenchajun) [inline conf/fluent-bit.conf:2]: I added a conf

File: controllers/clusterfluentbitconfig_controller.go [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		parserCfg, err := cfg.RenderParserConfig(sl, clusterParsers)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		cl := plugins.NewConfigMapLoader(r.Client, ns)
+		scripts, err := cfg.RenderLuaScript(cl, clusterFilters)
```
Comment (wanjunlei) [inline controllers/clusterfluentbitconfig_controller.go:128]: The script for cluster Lua filter and namespace Lua filter are in the different directories, how to handle this?

**Review by wanjunlei — CHANGES_REQUESTED** [review]

**Comment by benjaminhuo** [comment]
@wanjunlei @zhu733756 pls help to review this PR and the #167 to decide which method we should use

**Pull request closed without merging**